tests/asserts/final/publication/micka.py

```diff
@@ -0,0 +1,108 @@
+from datetime import date
+
+import crs as crs_def
+from layman import app, util as layman_util, settings, LaymanError
+from layman.common import bbox as bbox_util
+from layman.layer import db
+# from layman.layer.filesystem import gdal
+from layman.layer.geoserver import wms
+from layman.layer.micka import csw as csw_util
+# from layman.layer.prime_db_schema import table as prime_db_table
+from test_tools import process_client, assert_util
+
+METADATA_PROPERTIES = {
+    'abstract',
+    'extent',
+    'graphic_url',
+    'identifier',
+    'layer_endpoint',
+    'language',
+    'organisation_name',
+    'publication_date',
+    'reference_system',
+    'revision_date',
+    # 'spatial_resolution',
+    'title',
+    'wfs_url',
+    'wms_url',
+}
+
+
+def expected_values_in_metadata(workspace, publ_type, name, expected_values):
+    assert publ_type == process_client.LAYER_TYPE
+    with app.app_context():
+        md_dict = csw_util.get_metadata_comparison(workspace, name)
+
+    assert len(md_dict) == 1
+    md_record = next(iter(md_record for md_record in md_dict.values()))
+
+    assert_util.assert_same_values_for_keys(expected=expected_values,
+                                            tested=md_record,
+                                            )
+
+
+def correct_values_in_layer_metadata(workspace, publ_type, name, ):
+    assert publ_type == process_client.LAYER_TYPE
+    with app.app_context():
+        resp_json = process_client.get_workspace_layer_metadata_comparison(workspace, name,)
+        assert METADATA_PROPERTIES.issubset(set(resp_json['metadata_properties'].keys()))
+        for key, value in resp_json['metadata_properties'].items():
+            assert value['equal_or_null'] is True, f'key={key}, value={value}'
+
+        publ_info = layman_util.get_publication_info(workspace, publ_type, name)
```

Comment:
Move _get_property_values logic into get_template_path_and_values and use get_template_path_and_values to get expected metadata values for the test.